api/metadata_handler.go

```diff
@@ -35,8 +34,8 @@ func apiMetadataHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := shared.NewAppEngineContext(r)
 	client := shared.NewAppEngineAPI(ctx).GetHTTPClient()
 	logger := shared.GetLogger(ctx)
-	metadataURL := shared.MetadataArchiveURL
-	delegate := MetadataHandler{logger, client, metadataURL}
+	fetcher := webappMetadataFetcher{ctx: ctx, client: client, log: logger, url: shared.MetadataArchiveURL}
+	delegate := MetadataHandler{logger, fetcher}
 
 	// Serve cached with 5 minute expiry. Delegate to Metadata Handler on cache miss.
 	shared.NewCachingHandler(
```

Comment:
We shouldn't use CachingHandler any more, which adds yet another layer of memcache (keyed by URL) that's redundant now.